src/Microsoft.ComponentDetection.Detectors/nuget/NuGetPackageReferenceFrameworkAwareDetector.cs

```diff
@@ -0,0 +1,231 @@
+namespace Microsoft.ComponentDetection.Detectors.NuGet;
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using global::NuGet.Packaging.Core;
+using global::NuGet.ProjectModel;
+using global::NuGet.Versioning;
+using Microsoft.ComponentDetection.Contracts;
+using Microsoft.ComponentDetection.Contracts.Internal;
+using Microsoft.ComponentDetection.Contracts.TypedComponent;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+public class NuGetPackageReferenceFrameworkAwareDetector : FileComponentDetector
```

Comment:
We need to add the experimental interface here to safely validate it beforehand:

```suggestion
public class NuGetPackageReferenceFrameworkAwareDetector : FileComponentDetector, IExperimentalDetector
```